src/lib/core/a11y/focus-trap.ts

```diff
@@ -15,15 +17,44 @@ import {InteractivityChecker} from './interactivity-checker';
   selector: 'focus-trap',
   // TODO(jelbourn): move this to a separate file.
   template: `
-  <div tabindex="0" (focus)="focusLastTabbableElement()"></div>
+  <div *ngIf="active" tabindex="0" (focus)="focusLastTabbableElement()"></div>
   <div #trappedContent><ng-content></ng-content></div>
-  <div tabindex="0" (focus)="focusFirstTabbableElement()"></div>`,
+  <div *ngIf="active" tabindex="0" (focus)="focusFirstTabbableElement()"></div>`,
   encapsulation: ViewEncapsulation.None,
 })
-export class FocusTrap {
+export class FocusTrap implements AfterContentInit {
   @ViewChild('trappedContent') trappedContent: ElementRef;
 
-  constructor(private _checker: InteractivityChecker) { }
+  @Input()
+  get active(): boolean {
+    return this._active;
+  }
+  set active(val: boolean) {
+    this._active = val;
+    if (val && this._contentReady) {
+      this._ngZone.onMicrotaskEmpty.first().subscribe(() => {
```

Comment:
This one is used here and repeated down inside `ngAfterContentInit`. It might a good idea to move it to a private method like `waitForTabbableElement`.